### DISCUSSION TITLE
[SURVEY] How are you using Evidence?

### DISCUSSION THREAD

**Discussion by hughess** [evidence-dev/evidence#2384] [Announcements]
We would love to know which companies / organizations use Evidence and what they use it for. This helps us understand how we can make the product and experience better. 

It’s also incredibly helpful for open source projects to be able to showcase organizations that use them, as it contributes to wider adoption and trust in the project. If you’re able to support us by doing this, we’d really appreciate it.

If you’re using Evidence, tell us below.

Ideas for what to include:
- Which company / organization is using Evidence?
- What are you using it for?
- Is there any general feedback you'd like to provide about the product, the development experience, or the end user experience?

**Comment by dhoe** [comment]
I'm currently using it as an experiment with a non-profit (about ten employees) to see if it helps people to have their finances summarized for them; built some dashboards to show monthly financial performance. They would definitely have near zero budget for this. Feedback I got from the dashboard consumers is that it looks great. I find Evidence easy to use and well documented. I chose it because I'm a SQL person and if I can avoid the big BI tools I will.

**Comment by sandstrom** [comment]
Feedback:

Maybe this is possible, just browsed the tool quickly, but here are some thoughts:

- Great if there would be a CLI tool that we could run on internal networks, that gathers the data and build the reports, and then pushes these reports to the "hosting service". The hosting could be either self-hosted or your cloud.
  - We'd be happy to use the cloud since the reports themselves won't be that "secret", however, we wouldn't be willing to give your cloud software access to our servers (data storage), for security reasons.
- The only "build target" right now seems to be a static HTML site with JS. It would be neat if it also supported e.g. "dumb html", so basically no interactivity in the widgets (maybe turning them into static SVGs or PNGs).
- For graph inspiration, take a look at https://metricsgraphicsjs.org/line. I think they have some of the nicest / cleanest looking graphs around. Unfortunately the library itself isn't very popular, for no obvious reason.

**Comment by ItsMeBrianD** [comment]
> Great if there would be a CLI tool that we could run on internal networks, that gathers the data and build the reports, and then pushes these reports to the "hosting service". The hosting could be either self-hosted or your cloud.

Would this be somewhat akin to [Self-Hosted Runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners)?

**Comment by sandstrom** [comment]
Probably closer to the CLI tools that are popular, for example https://docs.sentry.io/product/cli/ and https://docs.readme.com/main/docs/rdme.

So basically, a push based mechanism where we can push updated data (based on the queries) to your cloud. The cloud would be the viewer, would handle access control for individual users, maybe inline comments and discussions on reports, and other user-to-user stuff.

In the scenario I'm thinking of, the markdown source would be version controlled by us (that's the idea with Evidence, no?) and we'd run a job on a production node that would periodically run the queries and push the result to your system using a CLI tool (or your API).

The benefit here is that if you get hacked, our database won't be. We'll pin your CLI tool with a SHA hash. Obviously we'll have to trust that you won't ship malicious code, but most of the scenarios where you get hacked wouldn't spill over into our systems.

It's also technically much easier to integrate in a manner where we run things inside our internal networks, gather the data and push them to you, than to open all the firewalls for your systems to have access into our databases.

Obviously this also depends a bit on which direction you want to go in. But if it's an honest open-source project, where you are willing to accept that some people will self-host your product, and you will earn money on a subset of customers, but not all, I'd probably draw the line like this:

## Cloud (paid)

- Collaboration (comments, discussion)
- SSO, SCIM, security, user controls
- Sharing of reports, sending out regular email reports
- Hosting of markdown source (for those that want it)
- Pulling data from sources (for those that want it)
- But give people the option of self-hosting the reports in Github, and push them to you via the CLI (via e.g. Github Actions runner).

## Open-source
- Markdown spec and language
- Components
- HTML generator
- CLI tool for uploading to your cloud or self-hosted
- All the other stuff you have open-source already

## Self-hosted

Can self-host using the open-source stuff. Make it actually possible to do, because even though many of them won't pay, getting widespread adoption would ultimately drive more people to your product. Self-hosters would need to handle collaboration, discussion, login, auth, sharing, etc on their own. Be honest about what's included here and what's not. But don't make it "too weak".


Also, you probably have already, but otherwise look at BSL or https://fsl.software/

**Comment by 0x1DKFA** [comment]
Hello,
First of all the new USQL looks great :) 
I was wondering if there's a way to embed the dashboard into an existing react app? Maybe by using iFrame? Just like how we can integrate the Superset dashboard using their SDK.

**Comment by archiewood** [comment]
There is. It's typically a custom setup using a JWT token.

There's a form you can fill out here (select Embedded) if you want to find out more: https://evidence.dev/cloud
